src/se/bitcraze/crazyfliecontrol/ble/BleLink.java

```diff
@@ -414,7 +414,7 @@ private void sendSplitPacket(CrtpPacket packet) {
         // controlbyte + payload (rest)
         byte[] secondPacket = new byte[20];
         secondPacket[0] = new ControlByte(false, pid, 0).toByte();
-        System.arraycopy(packet.getPayload(),19, secondPacket, 1, packet.getPayload().length-19);
```

Comment:
`packet.getPayload().length-19` should be `packet.getPayload().length-18` since 18 bytes from the payload are in the first packet, the rest in the second packet.

Reply:
Nice catch, i missed that one, it fixed an earlier problem i had with split package parameter setting not working. The weird thing is it only works when i debug and have a breakpoint in the function (although after a couple of resends through the resend queue). So it might be a thing with the rate packages are send out. since the breakpoint does introduce a delay.
The weird thing i cannot get the paramater set when it is running normally through the same sequence (so without the delay from debugging it).

Reply:
There are definitely some hidden dragons. I remember that debugging the queues was always a pain. Some of the code desperately needs a second pair of eyes. :grimacing:

Reply:
Fixed the off-by-one error in 28192a3.